src/hooks/useGridDimensions.ts

```diff
@@ -0,0 +1,41 @@
+import { useRef, useState, useLayoutEffect } from 'react';
+
+// https://github.com/microsoft/TypeScript/issues/37861
+interface ResizeObserverEntry {
+  contentRect: {
+    width: number;
+    height: number;
+  };
+}
+
+type ResizeObserver = new (callback: (entries: readonly ResizeObserverEntry[]) => void) => {
+  observe: (target: Element) => void;
+  disconnect: () => void;
+};
+
+export function useGridDimensions(): [React.RefObject<HTMLDivElement>, number, number] {
+  const gridRef = useRef<HTMLDivElement>(null);
+  const [gridWidth, setGridWidth] = useState(1);
+  const [gridHeight, setGridHeight] = useState(1);
+
+  useLayoutEffect(() => {
+    const { ResizeObserver } = window as typeof window & { ResizeObserver: ResizeObserver };
+
+    // don't break in jest/jsdom and browsers that don't support ResizeObserver
+    if (ResizeObserver == null) return;
```

Comment:
So in this case we render an empty grid? I wonder if we should throw an exception

Reply:
Almost.

![image](https://user-images.githubusercontent.com/567105/91182695-69306c80-e6e2-11ea-8805-e9ae410fc86b.png)

We could log something with `console.error`, but it'd spam jest tests as well.